scripts/check_no_panic_in_production.py

```
@@ -0,0 +1,169 @@
+            # The opening brace is at current depth; the scope is from
+            # depth+1 inwards. Pop it when depth drops back to current.
+            test_scope_stack.append(depth)
+        pending_test_attr = False if opens_scope or not is_attr_line else pending_test_attr
+
+        # Apply opens before checking panics, then closes after.
+        depth_before_line = depth
+        depth += opens
+
+        # Check this line for panics.
+        if PANIC_MACROS.search(stripped_line):
+            if ALLOW_MARKER not in raw_line:
```

Ruff (0.15.10)

[warning] Use a single `if` statement instead of nested `if` statements

Combine `if` statements using `and`

(SIM102)